src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -540,6 +543,32 @@ protected function SaveOrderNotificationDataToOrder($bOrderSend, $oMail, $sSendT
         $this->AllowEditByAll(false);
     }
 
+    protected function SetSendOrderNotificationState(bool $orderSend): void
+    {
+        if (true === TGlobal::IsCMSMode()) {
+            return;
+        }
+
+        $aData = $this->sqlData;
+        if (true === $orderSend) {
+            $aData = $this->AddOrderNotificationDataForOrderSuccess($aData);
+        } else {
+            $aData = $this->AddOrderNotificationDataForOrderNotDone($aData);
+        }
+
+        $this->LoadFromRow($aData);
```

Comment:
Does this data need to be loaded here (is it not up-to-date?)?

Reply:
after changeing the field state we need to update $this object